platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -176,26 +175,17 @@ public BaseEventUnhashedData getUnhashedData() {
      */
     @Override
     public EventDescriptor getDescriptor() {
-        if (descriptor == null) {
-            throw new IllegalStateException("Can not get descriptor until event has been hashed");
-        }
-        return descriptor;
+        return hashedData.getDescriptor();
     }
 
     /**
-     * Build the descriptor of this event. This cannot be done when the event is first instantiated, it needs to be
-     * hashed before the descriptor can be built.
+     * Ensure that the event descriptor has been built This cannot be done when the event is first instantiated, it
+     * needs to be hashed before the descriptor can be built.
      *
      * @throws IllegalStateException if the descriptor has already been built
      */
     public void buildDescriptor() {
```

Comment:
method is unused, and a duplicated of `getDescriptor`

Reply:
removed